mat64/triangular.go

```diff
@@ -0,0 +1,55 @@
+package mat64
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+var (
+	triangular *Triangular
+
+	_ Matrix  = triangular
```

Comment:
naive comment: isn't "cleaner" to just spell it out
`_ Matrix = (*Triangular)(nil)`

admittedly more verbose, but no "pollution" of the global scope.